packages/frontend-api/src/Model/Resolver/Image/ImagesResolver.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrontendApiBundle\Model\Resolver\Image;
+
+use Overblog\GraphQLBundle\Definition\Resolver\ResolverInterface;
+use Overblog\GraphQLBundle\Error\UserError;
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Shopsys\FrameworkBundle\Component\Image\Config\ImageConfig;
+use Shopsys\FrameworkBundle\Component\Image\Config\ImageEntityConfig;
+use Shopsys\FrameworkBundle\Component\Image\ImageFacade;
+use Shopsys\FrameworkBundle\Model\Category\Category;
+use Shopsys\FrameworkBundle\Model\Product\Product;
+
+class ImagesResolver implements ResolverInterface
+{
+    protected const IMAGE_PRODUCT_ENTITY = 'product';
+    protected const IMAGE_CATEGORY_ENTITY = 'category';
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Image\ImageFacade
+     */
+    protected $imageFacade;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Image\Config\ImageConfig
+     */
+    protected $imageConfig;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Domain\Domain
+     */
+    protected $domain;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Image\ImageFacade $imageFacade
+     * @param \Shopsys\FrameworkBundle\Component\Image\Config\ImageConfig $imageConfig
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     */
+    public function __construct(
+        ImageFacade $imageFacade,
+        ImageConfig $imageConfig,
+        Domain $domain
+    ) {
+        $this->imageFacade = $imageFacade;
+        $this->imageConfig = $imageConfig;
+        $this->domain = $domain;
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\Product $product
+     * @param string|null $type
+     * @param string|null $size
+     * @return array
+     */
+    public function resolveByProduct(Product $product, ?string $type, ?string $size): array
+    {
+        return $this->resolveByEntity($product, static::IMAGE_PRODUCT_ENTITY, $type, $size);
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Category\Category $category
+     * @param string|null $type
+     * @param string|null $size
+     * @return array
+     */
+    public function resolveByCategory(Category $category, ?string $type, ?string $size): array
+    {
+        return $this->resolveByEntity($category, static::IMAGE_CATEGORY_ENTITY, $type, $size);
+    }
+
+    /**
+     * @param object $entity
+     * @param string $entityName
+     * @param string|null $type
+     * @param string|null $size
+     * @return array
+     */
+    protected function resolveByEntity(object $entity, string $entityName, ?string $type, ?string $size): array
+    {
+        $resolvedImages = [];
+        $sizeConfigs = $this->getSizeConfigs($type, $size, $entityName);
+        $images = $this->imageFacade->getImagesByEntityIndexedById($entity, $type);
+
+        foreach ($images as $image) {
```

Comment:
Can we extract it into separate method? To accomplish something like `return $this->getResolvedImages($images, $sizeConfigs)`. (I am not sure if `resolved` is the right word)

Reply:
Done